### PR TITLE
feat(time-picker): create package

### PR DESCRIPTION
**Pull request by gloriliale**
✅ created a Time Picker package
 ✅ wrote tests for Time Picker
 ✅ made stories for Time Picker
- [x] addressed the seconds/milliseconds and API question

### CODE REVIEW

File: packages/paste-core/components/time-picker/__tests__/index.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    expect(formatReturnTime(TEST_TIME, 'hh:mm')).toEqual('01:30');
+    expect(formatReturnTime(TEST_TIME, 'HH:mm:ss')).toEqual('13:30:04');
+    expect(formatReturnTime(TEST_TIME, 'hh:mm:ss')).toEqual('01:30:04');
+    expect(formatReturnTime(TEST_TIME, 'HH:mm:ss.SSS')).toEqual('13:30:04.000');
+    expect(formatReturnTime(TEST_TIME, 'HH:mm:ss.SSS')).toEqual('13:30:04.000');
+    expect(formatReturnTime(TEST_TIME, 'hh:mm:ss.SSS aa')).toEqual('01:30:04.000 p.m.');
+    // testing midnight
+    expect(formatReturnTime('00:00', 'hh:mm A')).toEqual('12:00 AM');
+    // test for edge cases (passing in time+date value) to throw an error
+    expect(() => {
```
Comment (andipants12) [inline packages/paste-core/components/time-picker/__tests__/index.spec.tsx:31]: Have you considered breaking down this `it` statement? In my experience, it's easier to debug test failures when there are less `expect` statements within an `it` block. For example, we could move the edge cases and the expected thrown error statement to a new block.

File: packages/paste-core/components/time-picker/src/TimePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import type {InputProps} from '@twilio-paste/input';
+import {Input} from '@twilio-paste/input';
+
+export type TimePickerProps = Omit<InputProps, 'type' | 'insertBefore' | 'insertAfter' | 'step'>;
+
+// @ts-ignore to remove step from props so it is not used
```
Comment (andipants12) [inline packages/paste-core/components/time-picker/src/TimePicker.tsx:8]: What is the reported TS error here? I'm wondering if we can find a resolution rather than using `ts-ignore`.

File: packages/paste-core/components/time-picker/src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (andipants12) [inline packages/paste-core/components/time-picker/src/utils.ts:1]: Is this eslint error still being reported? I see that `date-fns` is in this component's `package.json` `dependencies`. AFAIK this error should only report if it is not listed.

File: packages/paste-core/components/input/src/Input.tsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import {styled, css} from '@twilio-paste/styling-library';
 import {InputBox} from '@twilio-paste/input-box';
 import {safelySpreadFormControlProps} from './utils';
 
-export type InputTypes = 'text' | 'email' | 'hidden' | 'number' | 'password' | 'search' | 'tel' | 'date';
+export type InputTypes = 'text' | 'email' | 'hidden' | 'number' | 'password' | 'search' | 'tel' | 'date' | 'time';
```
Comment (richbachman) [inline packages/paste-core/components/input/src/Input.tsx:7]: Could these be imported from input-box? That would save some duplication.

**Comment by nkrantz** [comment]
@TheSisb Test 123....

File: packages/paste-core/components/input/src/Input.tsx (hunk 4 above)
Comment (nkrantz) [reply packages/paste-core/components/input/src/Input.tsx:7]: Good catch! Done

File: packages/paste-core/components/time-picker/src/TimePicker.tsx (hunk 2 above)
Comment (nkrantz) [reply packages/paste-core/components/time-picker/src/TimePicker.tsx:8]: I added a better comment to try to address this! Also changed it to a `@ts-expect-error`. It's because we're omitting `step` from the type but still destructuring it from the time picker props.

File: packages/paste-core/components/time-picker/src/utils.ts (hunk 3 above)
Comment (nkrantz) [reply packages/paste-core/components/time-picker/src/utils.ts:1]: Good catch! Removed from `time-picker` and `date-picker`.

File: packages/paste-core/components/time-picker/__tests__/index.spec.tsx (hunk 1 above)
Comment (nkrantz) [reply packages/paste-core/components/time-picker/__tests__/index.spec.tsx:33]: Thanks for noticing this! I pulled out the last 2 expects in `time-picker` and `date-picker`.

**Review by nkrantz — APPROVED** [review]

**Review by richbachman — APPROVED** [review]
